x-pack/legacy/plugins/ml/server/routes/annotations.ts

```diff
@@ -0,0 +1,151 @@
+        ' created or are not accessible for the current user.',
+    })
+  );
+}
+/**
+ * Routes for annotations
+ */
+export function annotationRoutes({ xpackMainPlugin, router }: RouteInitialization) {
+  /**
+   * @apiGroup Annotations
+   *
+   * @api {post} /api/ml/annotations Gets annotations
+   * @apiName GetAnnotations
+   * @apiDescription Gets annotations.
+   *
+   * @apiParam {String[]} jobIds List of job IDs
+   * @apiParam {String} earliestMs
+   * @apiParam {Number} latestMs
+   * @apiParam {Number} maxAnnotations Max limit of annotations returned
+   *
+   * @apiSuccess {Boolean} success
+   * @apiSuccess {Object} annotations
+   */
+  router.post(
+    {
+      path: '/api/ml/annotations',
+      validate: {
+        body: schema.object({
+          jobIds: schema.arrayOf(schema.string()),
+          earliestMs: schema.number(),
+          latestMs: schema.number(),
+          maxAnnotations: schema.number(),
+        }),
+      },
+    },
+    licensePreRoutingFactory(xpackMainPlugin, async (context, request, response) => {
+      try {
+        const { getAnnotations } = annotationServiceProvider(context);
+        const resp = await getAnnotations(request.body);
+
+        return response.ok({
+          body: resp,
+        });
+      } catch (e) {
+        return response.customError(wrapError(e));
+      }
+    })
+  );
+
+  /**
+   * @apiGroup Annotations
+   *
+   * @api {put} /api/ml/annotations/index Index annotation
+   * @apiName IndexAnnotations
+   * @apiDescription Index the annotation.
+   *
+   * @apiParam {Object} annotation
+   * @apiParam {String} username
+   */
+  router.put(
+    {
+      path: '/api/ml/annotations/index',
+      validate: {
+        body: schema.object(indexAnnotationSchema),
+      },
+    },
+    licensePreRoutingFactory(xpackMainPlugin, async (context, request, response) => {
+      try {
+        const annotationsFeatureAvailable = await isAnnotationsFeatureAvailable(
+          context.ml!.mlClient.callAsCurrentUser
+        );
+        if (annotationsFeatureAvailable === false) {
+          throw getAnnotationsFeatureUnavailableErrorMessage();
+        }
+
+        const { indexAnnotation } = annotationServiceProvider(context);
+        const username = _.get(request, 'auth.credentials.username', ANNOTATION_USER_UNKNOWN);
+        const resp = await indexAnnotation(request.body, username);
+
+        return response.ok({
+          body: resp,
+        });
+      } catch (e) {
+        return response.customError(wrapError(e));
+      }
+    })
+  );
+
+  /**
+   * @apiGroup Annotations
+   *
+   * @api {delete} /api/ml/annotations/index Deletes annotation
+   * @apiName DeleteAnnotation
+   * @apiDescription Deletes specified annotation
+   *
+   * @apiParam {String} annotationId
+   */
+  router.delete(
+    {
+      path: '/api/ml/annotations/delete/{annotationId}',
+      validate: {
+        params: schema.object({ annotationId: schema.string() }),
```

Comment:
What do you think of moving all schema definitions to `new_platform/annotations_schema.ts` for consistency even if they are not as complex as the `indexAnnotationSchema` one?